src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java

```diff
@@ -562,22 +562,22 @@ public SscsPipCaseData getSscsPipCaseData() {
     }
 
     @JsonIgnore
-    public Optional<BenefitTypeEnum> getBenefitType() {
+    public Optional<Benefit> getBenefitType() {
         if (appeal != null && appeal.getBenefitType() != null && appeal.getBenefitType().getCode() != null) {
             try {
-                return Optional.of(BenefitTypeEnum.valueOf(appeal.getBenefitType().getCode().toUpperCase()));
-            } catch (IllegalArgumentException e) {
+                return Optional.of(Benefit.findBenefitByShortName(appeal.getBenefitType().getCode().toUpperCase()));
+            } catch (NullPointerException e) {
```

Comment:
Why we are checking null pointer exception here? Ideally we should check if null then do this or else
Here we are suppressing the any other null pointer exception, which I think its not good idea

Reply:
Done